src/zenml/stack_stores/local_stack_store.py

```diff
@@ -187,6 +187,94 @@ def register_stack_component(
             component.name,
         )
 
+    def update_stack_component(
+        self,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Update a stack component."""
+        components = self.__store.stack_components[component.type]
+        if component.name not in components:
+            raise StackComponentExistsError(
+                f"Unable to update stack component (type: {component.type}) "
+                f"with name '{component.name}': No existing stack component "
+                f"found with this name."
+            )
+
+        # write the component configuration file
+        component_config_path = self._get_stack_component_config_path(
+            component_type=component.type, name=component.name
+        )
+        utils.create_dir_recursive_if_not_exists(
+            os.path.dirname(component_config_path)
+        )
+        utils.write_file_contents_as_string(
+            component_config_path,
+            base64.b64decode(component.config).decode(),
+        )
+
+        # add the component to the stack store dict and write it to disk
+        components[component.name] = component.flavor
+        self._write_store()
+        logger.info(
+            "Updated stack component with type '%s' and name '%s'.",
+            component.type,
+            component.name,
+        )
+
+    def rename_stack_component(
+        self,
+        old_name: str,
+        component: StackComponentWrapper,
```

Comment:
woudln't it be enough to have the new name here actually instead of the `StackComponentWrapper`?

I also think that this method needs to first check if a component with the name already exists and raise an exception in that case

Reply:
100%. I hadn't thought to check for that. Will amend + add tests to guard against that case, too.